python/tank/platform/constants.py

```diff
@@ -175,6 +175,12 @@
 # the name of the shell engine
 SHELL_ENGINE = "tk-shell"
 
+# valid characters for a template key name
+TEMPLATE_KEY_NAME_REGEX = "[a-zA-Z_ 0-9\.]+"
+
+# a human readable explanation of the above. For error messages.
+VALID_TEMPLATE_KEY_NAME_DESC = "letters, numbers, underscore, space and period"
```

Comment:
As for this, since only TemplateKey is concerned by it, it feels premature to expose it this way, unless this is how we expose all errors strings which makes it easier for localizing strings?